src/coreclr/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -138,15 +144,32 @@ public static int OffsetToStringData
         [MethodImpl(MethodImplOptions.InternalCall)]
         public static extern bool TryEnsureSufficientExecutionStack();
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern object GetUninitializedObjectInternal(
+        public static object GetUninitializedObject(
             // This API doesn't call any constructors, but the type needs to be seen as constructed.
             // A type is seen as constructed if a constructor is kept.
             // This obviously won't cover a type with no constructor. Reference types with no
             // constructor are an academic problem. Valuetypes with no constructors are a problem,
             // but IL Linker currently treats them as always implicitly boxed.
             [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors)]
-            Type type);
+            Type type)
+        {
+            if (type is null)
+            {
+                throw new ArgumentNullException(nameof(type), SR.ArgumentNull_Type);
+            }
+
+            if (!(type is RuntimeType rt))
+            {
+                throw new SerializationException(SR.Format(SR.Serialization_InvalidType, type));
+            }
```

Comment:
Any benefit to moving the null check?
```suggestion
            if (type is not RuntimeType rt)
            {
                if (type is null)
                {
                    throw new ArgumentNullException(nameof(type), SR.ArgumentNull_Type);
                }
            
                throw new SerializationException(SR.Format(SR.Serialization_InvalidType, type));
            }
```